.github/workflows/BuildAndTest.yml

```diff
@@ -36,21 +49,41 @@ jobs:
           # If source files changed but packages didn't, rebuild from a prior cache.
           restore-keys: |
             ${{ runner.os }}-nextjs-${{ hashFiles('**/package-lock.json') }}-
-          
-      - name: Yarn install
-        run: yarn install
-        
+
+      # Building the application
       - name: Yarn build
         run: yarn build
-      
-      - name: Yarn lint
-        run: yarn lint
 
-      - name: Run unit tests
-        run: yarn unit-test
+      # Installing Playwright's browsers
+      - name: Install Playwright's browsers
+        run: npx playwright install --with-deps
 
-      - name: Run component tests
-        run: yarn component-test:ci
-        
-      - name: Run E2E tests
-        run: yarn e2e-test:ci
\ No newline at end of file
+      # Running application tests
+      - name: Yarn test
+        run: yarn test
+
+      # Running E2E screenshots check
+      - name: E2E check
+        run: yarn e2e-check
+
+      # Committing E2E screenshot changes
+      - name: Commit E2E changes
```

Comment:
We need permissions flag at the top to do this

Reply:
Added write `permission:contents` to the job. That was enough to get it working on my test repo